### DISCUSSION TITLE
Wican automatically polling Can IDs for use e.g. with EVCC

### DISCUSSION THREAD

**Discussion by naseweiss99** [meatpiHQ/wican-fw#152] [Ideas]
Hi ,
when using a WICAN OBD adapter with EEVC to get the SoC of our Citroen e-C4, I have only a very smal time windows to get the data via my home Wifi into the EVCC application.

The time starts, when the car with the Wican is near enough to log into the Wifi and stops when ignition is turned of.

To avoid the use of an additional software/application it would be great to automatically generate the SoC poll telegramm and transmit the recieved value via MQTT to the relevant application.
The time intervall for this should be configurable. The IDs/PIDs polled & pushed, too.

I believe, this would create an additional demand for new Wican interfaces, as the integration in applications like EVCC would be even easier  - and Wican knows best when it is logged in into the local WIFI and ignition is still on.

many thanks

**Comment by meatpiHQ** [comment]
@naseweiss99 I've been working on this feature for a while now. Hopefully we'll have something to test in couple of weeks.

It would be great if you post your car PID request and response, I'll add it to the custom PIDs.

**Reply by naseweiss99** [reply]
Hello, that sounds really great.

this is the request send via MQTT publish:

{"bus":"0","type":"tx","frame":[{"id":1716,"dlc":8,"rtr":false,"extd":false,"data":[3,34,212,16,170,170,170,170]}]}


This is for the answer:

![image](https://github.com/meatpiHQ/wican-fw/assets/155883521/d49d848d-c6d4-4c76-9a5a-a919d32a57ea)


many thanks

**Reply by meatpiHQ** [reply]
@naseweiss99 you can try this firmware, I've added the autopid request in a new Automate tab.

https://github.com/meatpiHQ/wican-fw/releases/tag/v3.11Beta

**Reply by naseweiss99** [reply]
Sorry for the late reply, summer vacation :-)

I will check early next week - and provide feedback.

Thanks

**Reply by naseweiss99** [reply]
Hi,
thank first of all for the fast implementation.

I have to admit, I am a bit struggling to understand the fields.

My pull command is via MQTT-Publish: {"bus":"0","type":"tx","frame":[{"id":1716,"dlc":8,"rtr":false,"extd":false,"data":[3,34,212,16,170,170,170,170]}]}

Do I need to put this into "initialisation" ?

The answer from the WICAN should be 
![image](https://github.com/user-attachments/assets/350cb0ba-74c5-4b79-bfa8-abac1b355cbb)


What to put where ? 
![image](https://github.com/user-attachments/assets/5014127f-504d-4b6e-93db-20eaecde934e)

this was one of my trials but it was not working.
![image](https://github.com/user-attachments/assets/d89918d4-ceaa-477f-bf92-5babe2f65d9f)

My mqqt-server is 192.168.178.40 - with port 1883.

many thanks

**Reply by meatpiHQ** [reply]
I suggest you connect to your vehicle using car scanner and log the communication. This will tell us the initialisation. 

The initialisation is not in json format, it's AT commands to setup the ELM327.

**Reply by meatpiHQ** [reply]
To clarify use carscanner logs not wican "elm327 logging".

**Comment by naseweiss99** [comment]
I have to admit, I am not quite getting the point.
If I publish via MQTT - with " {"bus":"0","type":"tx","frame":[{"id":1716,"dlc":8,"rtr":false,"extd":false,"data":[3,34,212,16,170,170,170,170]}]}" 
Wican answers and the data is formated by the MQTT Web-frontend and send to the MQTT server  - what additionial thing is needed then ?

On the other hand I don't find the menu point for getting the logged carscanner data within carscanner - other the real values.
<img width="488" alt="image" src="https://github.com/user-attachments/assets/c7cdbaea-f444-480d-8cd0-9c04c3a67749">

I appreciate the effort put into this Software, however I would appreciate a bit of documentation, too.
Sorry, but I don't know what to enter where and why.

**Reply by meatpiHQ** [reply]
Please try this
**PID:** 22D410
**Expression:** B4/2
<img width="873" alt="image" src="https://github.com/user-attachments/assets/2e9e4626-77f2-4c86-aa31-a509dbfd75a9">
